--- openssh-10.0p1.orig/includes.h	2025-08-01 14:34:27.000000000 +0000
+++ openssh-10.0p1/includes.h	2025-08-01 15:00:23.000000000 +0000
@@ -62,6 +62,9 @@
 #ifdef HAVE_LOGIN_H
 # include <login.h>
 #endif
+#ifdef HAVE_SHADOW_H
+# include <shadow.h>
+#endif
 
 #ifdef HAVE_UTMP_H
 #  include <utmp.h>
--- openssh-10.0p1.orig/openbsd-compat/openbsd-compat.h	2025-08-01 14:34:27.000000000 +0000
+++ openssh-10.0p1/openbsd-compat/openbsd-compat.h	2025-08-01 15:00:23.000000000 +0000
@@ -107,7 +107,7 @@
 #endif
 
 #ifndef HAVE_RRESVPORT_AF
-int rresvport_af(int *alport, sa_family_t af);
+int rresvport_af(int *alport, int af);
 #endif
 
 #ifndef HAVE_STRLCPY
--- openssh-10.0p1.orig/openbsd-compat/rresvport.c	2025-08-01 14:34:27.000000000 +0000
+++ openssh-10.0p1/openbsd-compat/rresvport.c	2025-08-01 15:00:23.000000000 +0000
@@ -55,7 +55,7 @@
 #endif
 
 int
-rresvport_af(int *alport, sa_family_t af)
+rresvport_af(int *alport, int af)
 {
 	struct sockaddr_storage ss;
 	struct sockaddr *sa;
